src/jdk.jpackage/windows/classes/jdk/jpackage/internal/WinMsiBundler.java

```diff
@@ -197,8 +197,8 @@ public WinMsiBundler() {
                 Map.entry("ui.wxf", new WixUiFragmentBuilder())
         ).map(e -> {
             e.getValue().setOutputFileName(e.getKey());
-            return e.getValue();
-        }).collect(Collectors.toList());
+            return (WixFragmentBuilder) e.getValue();
```

Comment:
Why this explicit cast is needed here?

Reply:
The explicit cast is needed because `Stream.toList()` is invariant, but the stream is producing a covariant type (by way of the type inferencer and `Stream.of(..)` accepting two subclasses of `WixFragmentBuilder`). We need to introduce a hint to bring the type back to the invariant form. On second thought I'm going to make this explicit type argument instead so it's clear that the casting isn't part to what the code is otherwise doing.